core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/edit-ema-palette/store/edit-ema-palette.store.ts

```diff
@@ -0,0 +1,289 @@
+import { ComponentStore, tapResponse } from '@ngrx/component-store';
+import { EMPTY, Observable, forkJoin } from 'rxjs';
+
+import { Injectable, inject } from '@angular/core';
+
+import { catchError, map, switchMap, tap } from 'rxjs/operators';
+
+import {
+    DotContentTypeService,
+    DotESContentService,
+    DotPropertiesService
+} from '@dotcms/data-access';
+import {
+    DotCMSContentlet,
+    DotCMSContentType,
+    DotContainerStructure,
+    DotPageContainerStructure
+} from '@dotcms/dotcms-models';
+
+export enum PALETTE_TYPES {
+    CONTENTTYPE = 'CONTENTTYPE',
+    CONTENTLET = 'CONTENTLET'
+}
+
+export enum EditEmaPaletteStoreStatus {
+    LOADING = 'LOADING',
+    LOADED = 'LOADED',
+    ERROR = 'ERROR'
+}
+
+export const PALETTE_PAGINATOR_ITEMS_PER_PAGE = 20;
+
+export interface DotPaletteState {
+    contentlets: {
+        filter: {
+            query: string;
+            contentTypeVarName: string;
+        };
+        items: DotCMSContentlet[];
+        totalRecords: number;
+        itemsPerPage: number;
+    };
+    contenttypes: {
+        filter: string;
+        items: DotCMSContentType[];
+    };
+    status: EditEmaPaletteStoreStatus;
+    currentPaletteType: PALETTE_TYPES;
+    allowedTypes: string[];
+    currentContentType?: string;
+}
+
+@Injectable()
+export class DotPaletteStore extends ComponentStore<DotPaletteState> {
+    private readonly dotConfigurationService = inject(DotPropertiesService);
+
+    constructor(
+        private dotContentTypeService: DotContentTypeService,
+        private dotESContentService: DotESContentService
+    ) {
+        super({
+            contentlets: {
+                items: [],
+                filter: { query: '', contentTypeVarName: '' },
+                totalRecords: 0,
+                itemsPerPage: PALETTE_PAGINATOR_ITEMS_PER_PAGE
+            },
+            contenttypes: { items: [], filter: '' },
+            status: EditEmaPaletteStoreStatus.LOADING,
+            currentPaletteType: PALETTE_TYPES.CONTENTTYPE,
+            allowedTypes: []
+        });
```

Comment:
can we lazy load this store?

Reply:
Not for now, we have a strange issue with lazy init in this store